src/rsz/src/RepairDesign.cc

```diff
@@ -118,6 +122,80 @@ void RepairDesign::repairDesign(double max_wire_length,
                           repaired_net_count);
 }
 
+void RepairDesign::performEarlySizingRound(float gate_gain,
+                                           float buffer_gain,
+                                           int& repaired_net_count)
+{
+  // keep track of user annotations so we don't remove them
+  std::set<std::pair<Vertex*, int>> slew_user_annotated;
+
+  // We need to override slews in order to get good required time estimates.
+  for (int i = resizer_->level_drvr_vertices_.size() - 1; i >= 0; i--) {
+    Vertex* drvr = resizer_->level_drvr_vertices_[i];
+    for (auto rf : {RiseFall::rise(), RiseFall::fall()}) {
+      if (!drvr->slewAnnotated(rf, min_) && !drvr->slewAnnotated(rf, max_)) {
+        sta_->setAnnotatedSlew(drvr,
+                               resizer_->tgt_slew_corner_,
+                               sta::MinMaxAll::all(),
+                               rf->asRiseFallBoth(),
+                               resizer_->tgt_slews_[rf->index()]);
+      } else {
+        slew_user_annotated.insert(std::make_pair(drvr, rf->index()));
+      }
+    }
+  }
+  findBufferSizes();
+
+  sta_->searchPreamble();
+  search_->findAllArrivals();
+
+  for (int i = resizer_->level_drvr_vertices_.size() - 1; i >= 0; i--) {
+    Vertex* drvr = resizer_->level_drvr_vertices_[i];
+    Pin* drvr_pin = drvr->pin();
+    Net* net = network_->isTopLevelPort(drvr_pin)
+                   ? network_->net(network_->term(drvr_pin))
+                   : network_->net(drvr_pin);
+    dbNet* net_db = db_network_->staToDb(net);
+    search_->findRequireds(drvr->level() + 1);
+
+    if (net && !resizer_->dontTouch(net) && !net_db->isConnectedByAbutment()
+        && !sta_->isClock(drvr_pin)
+        // Exclude tie hi/low cells and supply nets.
+        && !drvr->isConstant()) {
+      float fanout, max_fanout, fanout_slack;
+      sta_->checkFanout(drvr_pin, max_, fanout, max_fanout, fanout_slack);
+      max_fanout = 10.0f;
```

Comment:
Why 10 and ignore the cell or design limit?

Reply:
This is a value which has worked well. Can we insert a flow default of max_fanout=10 or 20 after I remove this hard-coded value?

Reply:
like this: https://github.com/The-OpenROAD-Project-private/OpenROAD-flow-scripts/commit/c75a1d92b0e2c9c51000042fa5045e596fe7114e

Reply:
There are many flows out there.  We can add it to ORFS but if this really is the best thing they maybe it is better to keep it in OR.  We can always add a flag if needed.

Reply:
I've removed the override. I think we can merge this and revisit later. We don't need to enable gain-based buffering in the flow right away